test/com/google/enterprise/adaptor/ad/AdAdaptorTest.java

```diff
@@ -1425,6 +1419,44 @@ public void testFakeAdaptorGetDocIds() throws Exception {
   }
 
   @Test
+  public void testMultipleFullGroupPushes() throws Exception {
+    MockLdapContext ldapContext = defaultMockLdapContext();
+    AdAdaptor adAdaptor = new FakeAdaptor(ldapContext);
+    RecordingDocIdPusher pusher = new RecordingDocIdPusher();
+    Map<String, String> configEntries = defaultConfig();
+    pushGroupDefinitions(adAdaptor, configEntries, pusher, /*fullPush=*/ true,
+        /*init=*/ true);
+
+    HashMap<GroupPrincipal, Collection<Principal>> goldenGroups =
+        new HashMap<GroupPrincipal, Collection<Principal>>();
+    Principal everyone = new GroupPrincipal("Everyone", "Default");
+    goldenGroups.put((GroupPrincipal) everyone, new ArrayList<Principal>());
+    goldenGroups.put(new GroupPrincipal("Authenticated Users@NT Authority",
+        "Default"), Arrays.asList(everyone));
+    goldenGroups.put(new GroupPrincipal("Interactive@NT Authority",
+        "Default"), Arrays.asList(everyone));
+    assertEquals(goldenGroups, pusher.getGroupDefinitions());
+
+    // Add the sam group.
```

Comment:
This is a little squirrelly. Is this just to avoid even more machinery to specify which groups to add to the context?

Reply:
And mostly because I didn't really understand all the machinations mockLdapContextForMakeDefs() was doing on top of the the defaultMockLdapContext(), so I just split that stuff out to be used here and in mockLdapContextForMakeDefs.